cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/validator/matcher/PatternMatcher.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2016 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.cloudslang.lang.compiler.validator.matcher;
+
+import java.util.regex.Pattern;
+
+/**
+ * @author Bonczidai Levente
+ * @since 8/30/2016
+ */
+public abstract class PatternMatcher {
+    private final Pattern pattern;
+
+    public PatternMatcher(Pattern pattern) {
+        this.pattern = pattern;
+    }
+
+    public boolean matchesEndToEnd(String input) {
```

Comment:
`matches `is a good name, following the Java convention, I should say that I don't necessarily am a fan of wrapping `Pattern`s. The typical use of a Pattern is to compile the regex once and then reuse it. Maybe check if we do need this dynamic `Pattern`s and if not replace with static `Pattern`s.